Improve error messages

When we get an error, we see the less than helpful `Failed to execute the build`.

I think functions like buildAndTag() need to display the underlying cause :

`
throw new MojoExecutionException("Failed to execute the build: " + ex.getCause());
`

Here is an example.

Before -

```
[ERROR] F8: Failed to execute the build: io.fabric8.maven.core.service.Fabric8ServiceException: Error while trying to build the image
```

After -

```

```

An example I see is -

```
[ERROR] F8: Failed to execute the build: io.fabric8.maven.core.service.Fabric8ServiceException: Error while trying to build the image
```

With use getCause() -

```
[ERROR] F8: Failed to execute the build: io.fabric8.maven.core.service.Fabric8ServiceException: Error while trying to build the image org.apache.maven.plugin.MojoExecutionException: Failed to create assembly for docker image  (with mode 'dir'): Error creating assembly archive docker: Problem copying files : /Users/plord/workspace/pipelinetest/ams/target/docker/ams/1.6.0-SNAPSHOT/build/maven/./tibco/sb-cep/10.6/ams/3rdparty/linux/nicstat/1.22_x86_64/nicstat.c (Permission denied).
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@ After this we will switch probably to real [Semantic Versioning 2.0.0](http://se
 * Fix #1755: Spring boot enricher does not produce a proper heath check and liveness check path when "/" is used.
 * Feature: Check maven.compiler.target property for base image detection.
 * Fix: Enrichers should resolve relative paths against project directory, not working directory
+* Fix #1769: Display root cause of failure if available
 
 ### 4.3.1 (18-10-2019)
 * Updated Kubernetes client to 4.6.1

--- a/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/BuildMojo.java
+++ b/plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/BuildMojo.java
@@ -296,7 +296,11 @@ public class BuildMojo extends io.fabric8.maven.docker.BuildMojo {
             fabric8ServiceHub.getBuildService().build(imageConfig);
 
         } catch (Exception ex) {
-            throw new MojoExecutionException("Failed to execute the build: " + ex);
+            if (ex.getCause() != null) {
+                throw new MojoExecutionException("Failed to execute the build: " + ex.getCause());
+            } else {
+                throw new MojoExecutionException("Failed to execute the build: " + ex);
+            }
         }
     }
 

